### DISCUSSION TITLE
Interpolater Support for FabArray<BaseFab<Vector<Real>>>   instead of MultiFab

### DISCUSSION THREAD

**Discussion by t-arshad** [AMReX-Codes/amrex#3518] [Q&A]
If I am dealing with a MultiFab that contains several variables (nComp>1) the variables are stored at far-off locations on the RAM.
Rather I want a container where different variables are stored close to one another for efficiency.
I can always make a FabArray<BaseFab<Vector<Real>>> but then coarse to fine interpolation is a problem as all interpolaters have a "interp" function that only accepts FArrayBox as its argument. Which means only MultiFabs can be interpolated.

Is there something I am missing? Or is writing a custom interpolator the only way to go?

Even if writing a customer Interpolater is the way to go. Class Interpolater.H has a pure virtual function "interp" that takes FArrayBox as an argument. Do I need to override this function? and overload it with a function that takes BaseFab<Vector<Real>> as an argument?

**Comment by WeiqunZhang** [comment]
Are you talking about `typename Interp` in `FillPatchSingleLevel` and `FillPatchTwoLevels`? These two functions are function template. So as long as it has a function named `interp` that takes `MF[MFIter] it will work. So they could have nothing to do with amrex::Interpolater and FArrayBox.

> Rather I want a container where different variables are stored close to one another for efficiency.

It would be bad for vectorization and GPU memory access efficiency.
